lib/auth/webauthncli/fido2_test.go

```diff
@@ -1016,6 +1017,124 @@ func TestFIDO2Login_u2fDevice(t *testing.T) {
 	assert.NoError(t, err, "FIDO2Login errored")
 }
 
+// TestFIDO2Login_u2fDeviceNotRegistered tests assertions with a non-registered
+// U2F device plugged.
+//
+// U2F devices error immediately when not registered, which makes their behavior
+// distinct from FIDO2 and requires additional logic to be correctly handled.
+//
+// This test captures an U2F assertion regression.
+func TestFIDO2Login_u2fDeviceNotRegistered(t *testing.T) {
```

Comment:
```suggestion
func TestFIDO2LoginU2FDeviceNotRegistered(t *testing.T) {
```

Would love to see us stop putting `_` in test names and just name them like we would any other func.

Reply:
I tend to name tests as "TestFunc_scenario" or "TestFunc_scenario_outcome". The idea is to make clear what is under test clear, while also specifying the scenario when it's relevant. It's similar to the Example format explained here: https://pkg.go.dev/testing#hdr-Examples.

I'm sure we can find other Go examples, although I don't think the original source that pushed me to it is available. You can find similar ideas from other languages as well (for example, this SO thread: https://stackoverflow.com/a/1594049).

I don't think it's as bad at looks, plus it beats the few tests where the target of test is otherwise not clearly specified. Happy to chat more about this, or even drop it if y'all it's not a good practice.

Reply:
I know you tend to name things this way, and it hasn't grown on me one bit over the years.

Tests are just functions, and we're explicitly breaking common conventions for how to name functions (from _Effective Go_ and _CodeReviewComments_) by creating new [unwritten] rules here.

I don't see why you need to leverage some special convention to make it clear what scenario is being tested - you can already do that with a well named function and clear godoc (which should apply to both test functions and non-test functions!)

Just my 2c - take it or leave it, you won't hurt my feelings either way :-)

Reply:
Fair enough. Thanks for the explanation.